featurebyte/feast/infra/offline_stores/spark_thrift.py

```diff
@@ -0,0 +1,353 @@
+                    quoted=True,
+                )
+            )
+        expr = (
+            expressions.Select(expressions=select_fields)
+            .from_(inner_expr.subquery())
+            .where(
+                expressions.EQ(
+                    this=quoted_identifier("feast_row_"),
+                    expression=expressions.Literal(this=1, is_string=False),
+                )
+            )
+        )
+        query = sql_to_string(expr, source_type=SourceType.SPARK)
+
+        return SparkThriftRetrievalJob(
+            db_session=db_session,
+            query=query,
+            config=config,
+            full_feature_names=False,
+            on_demand_feature_views=None,
+        )
+
+    @staticmethod
+    def pull_all_from_table_or_query(
+        config: RepoConfig,
+        data_source: DataSource,
+        join_key_columns: List[str],
+        feature_name_columns: List[str],
+        timestamp_field: str,
+        start_date: datetime,
+        end_date: datetime,
+    ) -> RetrievalJob:
+        raise NotImplementedError()
+
+    @staticmethod
+    def get_historical_features(
+        config: RepoConfig,
+        feature_views: List[FeatureView],
+        feature_refs: List[str],
+        entity_df: Union[pd.DataFrame, str],
+        registry: BaseRegistry,
+        project: str,
+        full_feature_names: bool = False,
+    ) -> RetrievalJob:
+        raise NotImplementedError()
+
+    @staticmethod
+    def offline_write_batch(
+        config: RepoConfig,
+        feature_view: FeatureView,
+        table: pyarrow.Table,
+        progress: Optional[Callable[[int], Any]],
+    ) -> None:
+        raise NotImplementedError()
+
+
+class SparkThriftRetrievalJob(RetrievalJob):
+    """
+    Spark Thrift Server retrieval job
+    """
+
+    def __init__(
+        self,
+        db_session: BaseSparkSession,
+        query: str,
+        full_feature_names: bool,
+        config: RepoConfig,
+        on_demand_feature_views: Optional[List[OnDemandFeatureView]] = None,
+        metadata: Optional[RetrievalMetadata] = None,
+    ):
+        super().__init__()
+        self.db_session = db_session
+        self.query = query
+        self._full_feature_names = full_feature_names
+        self._on_demand_feature_views = on_demand_feature_views or []
+        self._metadata = metadata
+        self._config = config
+
+    @property
+    def full_feature_names(self) -> bool:
+        return self._full_feature_names
+
+    @property
+    def on_demand_feature_views(self) -> List[OnDemandFeatureView]:
+        return self._on_demand_feature_views
+
+    def _to_df_internal(self, timeout: Optional[int] = None) -> pd.DataFrame:
+        """
+        Return dataset as Pandas DataFrame synchronously
+
+        Parameters
+        ----------
+        timeout: Optional[int]
+            Timeout in seconds
+
+        Returns
+        -------
+        pd.DataFrame
+        """
+        statements = self.query.split("---EOS---")
```

Comment:
Is the query here generated by us? Does it contain the `---EOS---` string?

Reply:
Oh indeed. I forgot to change this part - we won't generate multi-line statements